ocaml/otherlibs/systhreads/st_stubs.c

```diff
@@ -194,27 +243,35 @@ static void save_runtime_state(void)
 {
   if (Caml_state->in_minor_collection)
     caml_fatal_error("Thread switch from inside minor GC");
-  CAMLassert(This_thread != NULL);
-  caml_thread_t this_thread = This_thread;
-  this_thread->current_stack = Caml_state->current_stack;
-  this_thread->c_stack = Caml_state->c_stack;
-  this_thread->gc_regs = Caml_state->gc_regs;
-  this_thread->gc_regs_buckets = Caml_state->gc_regs_buckets;
-  this_thread->exn_handler = Caml_state->exn_handler;
-  this_thread->async_exn_handler = Caml_state->async_exn_handler;
-  this_thread->local_roots = Caml_state->local_roots;
-  this_thread->local_arenas = caml_get_local_arenas(Caml_state);
-  this_thread->backtrace_pos = Caml_state->backtrace_pos;
-  this_thread->backtrace_buffer = Caml_state->backtrace_buffer;
-  this_thread->backtrace_last_exn = Caml_state->backtrace_last_exn;
+  /* CR zqian: we save to [active_thread] instead of [this_thread]. I believe
+  they are equivalent here, but I think use [active_thread] is easier to
+  understand, also follows the systhreads4 behavior. */
```

Comment:
Yes, Active_thread makes more sense to me too, but I agree they should be the same here: to call `save_runtime_state`, the current thread should be active. Maybe add a `CAMLassert` that they're equal?